gallery/views.py

```diff
@@ -0,0 +1,119 @@
+from django.http.response import HttpResponse
+from django.views.generic import DetailView, ListView, FormView
+from django.urls import reverse
+from django.contrib import messages
+from django.shortcuts import redirect
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.templatetags.static import static
+from gallery.models import Image, Album
+from gallery.forms import ImageCreateForm
+from gallery import settings
+
+
+class GallerySettingsMixin(object):
+    """ Apply Gallery's Settings to a view """
+
+    def get_context_data(self, **kwargs):
+        """ Make settings available the template """
+        context = super(GallerySettingsMixin, self).get_context_data(**kwargs)
+        context['logo_path'] = settings.GALLERY_LOGO_PATH
+        context['gallery_title'] = settings.GALLERY_TITLE
+        context['hdpi_factor'] = settings.GALLERY_HDPI_FACTOR
+        context['image_margin'] = settings.GALLERY_IMAGE_MARGIN
+        context['footer_info'] = settings.GALLERY_FOOTER_INFO
+        context['footer_email'] = settings.GALLERY_FOOTER_EMAIL
+        context['theme_css_path'] = static('gallery/css/themes/{}.css'.format(settings.GALLERY_THEME_COLOR))
+
+        return context
+
+
+class ImageView(GallerySettingsMixin, DetailView):
+    model = Image
+
+    def get_context_data(self, **kwargs):
+        context = super(ImageView, self).get_context_data(**kwargs)
+        context['album_images'] = []
+        context['apk'] = self.kwargs.get('apk')
+
+        context['next_image'] = None
+        context['previous_image'] = None
+
+        # If there is an album in the context, look up the images in it
+        if context['apk']:
+            context['album'] = Album.objects.get(pk=context['apk'])
+            images = context['album'].images.all()
+            album_images = sorted(images, key=lambda i: i.date_taken)
+            context['album_images'] = album_images
+            for i in range(len(album_images)):
+                if self.object.pk == album_images[i].pk:
+                    if i > 0:
+                        context['previous_image'] = album_images[i - 1]
+                    if i < len(album_images) - 1:
+                        context['next_image'] = album_images[i + 1]
+        else:
+            # Look for albums this image appears in
+            context['albums'] = self.object.image_albums.all()
+
+        return context
+
+
+class ImageList(GallerySettingsMixin, ListView):
+    model = Image
+
+    def get_queryset(self):
+        # Order by newest first
+        return super(ImageList, self).get_queryset().order_by('-pk')
+
+
+class ImageCreate(GallerySettingsMixin, LoginRequiredMixin, FormView):
```

Comment:
Can we get a perm check here (not only loginRequied)

Reply:
I assumed it was either asking for superuser or some custom permission, but I never checked... doh, will fix for sure